ci-operator/step-registry/upi/gcp/rhel/pre/upi-gcp-rhel-pre-ref.yaml

```diff
@@ -0,0 +1,32 @@
+ref:
+  as: upi-gcp-rhel-pre
+  from: libvirt-installer
+  grace_period: 10m
+  commands: upi-gcp-rhel-pre-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
+    - name: HOME
+      default: /tmp/secret
+    - name: NSS_WRAPPER_PASSWD
+      default: /tmp/secret/passwd
+    - name: NSS_WRAPPER_GROUP
+      default: /tmp/secret/group
+    - name: NSS_USERNAME
+      default: rhel8user
+    - name: NSS_GROUPNAME
+      default: rhel8user
+    - name: GOOGLE_COMPUTE_IMAGE_FAMILY
+      default: rhel-8
+    - name: GOOGLE_COMPUTE_IMAGE_NAME
+      default: ''
+    - name: GOOGLE_COMPUTE_IMAGE_PROJECT
+      default: rhel-cloud
+    - name: GOOGLE_COMPUTE_MACHINE_TYPE
```

Comment:
I am wondering if this should be simpler and choose between x86 and ARM only, instead of going down to the specific machine type.

Reply:
+1
I like the parametrization but for setting up jobs it'd be best to have simple choices (and obvious at first glance).
Details would be hidden in `upi-gcp-rhel-pre-commands.sh` or as defaults in `upi-gcp-rhel-pre-ref.yaml`, but in `openshift-microshift-main.yaml` we'd just have smth like:
```
- as: e2e-openshift-conformance-reduced
  cluster: build02
  run_if_changed: ^(assets/.*|cmd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
  steps:
    cluster_profile: gcp
    workflow: openshift-microshift-e2e-openshift-conformance-reduced
    env:
      "ARCH": "x86"                # { "x86" | "arm" } 
      "RHEL": "8"                  # { "8" | "9" }
```

Reply:
Agreed, that'll be a nicer abstraction.  Will readjust.

Reply:
Actually, the way I'd envisioned this was to add rhel8[amd64,arm64] and rhel9[amd64,arm64] step refs that set these variables. This way we wouldn't have to come up with our own env var API and instead let it be embedded in the CI's existing APIs.